src/features/accuse/index.tsx

```diff
@@ -0,0 +1,142 @@
+import { Button, Flex, FormControl, FormLabel, Input, Text } from '@chakra-ui/react';
+import { useAccuse } from 'hooks/thirdPartyFacet/useAccuse';
+import React, { useEffect, useState } from 'react';
+import { useAccount } from 'wagmi';
+import { PrivateKeyInput } from './PrivateKeyInput';
+import { useAccusedWallets } from './useAccusedWallets';
+
+// 0x7a034a19af32e657f4462dd1f1cd61205f2ba21a8ffcaafa6fb968defb4e4be8
+export function Accuse() {
+  const [sarcophagusId, setSarcophagusId] = useState('');
+  const [archaeologistPrivateKeys, setArchaeologistPrivateKeys] = useState<string[]>(['']);
+  const [paymentAddressOverride, setPaymentAddressOverride] = useState('');
+
+  const { address: paymentAddress } = useAccount();
+
+  // Sign the messages using the private keys whenever the form is provided valid values
+  const { signatures, publicKeys } = useAccusedWallets(
+    sarcophagusId,
+    archaeologistPrivateKeys,
+    paymentAddress
+  );
+
+  const { accuse, isError, isAccusing, isLoading, isSuccess, isEnabled } = useAccuse(
+    sarcophagusId,
+    publicKeys,
+    signatures,
+    paymentAddress
+  );
+
+  function handleChangeSarcophagusId(e: React.ChangeEvent<HTMLInputElement>) {
+    setSarcophagusId(e.target.value);
+  }
+
+  function handleAddPrivateKey() {
+    setArchaeologistPrivateKeys(prev => [...prev, '']);
+  }
+
+  function handleRemovePrivateKey(index: number) {
+    const newArchaeologistPrivateKeys = [...archaeologistPrivateKeys];
+    newArchaeologistPrivateKeys.splice(index, 1);
+
+    // if there are no archaeologist private keys left, add an empty one
+    if (newArchaeologistPrivateKeys.length === 0) {
+      newArchaeologistPrivateKeys.push('');
+    }
+
+    setArchaeologistPrivateKeys(newArchaeologistPrivateKeys);
+  }
+
+  function handleChangePrivateKey(e: React.ChangeEvent<HTMLTextAreaElement>, index: number) {
+    const newArchaeologistPrivateKeys = [...archaeologistPrivateKeys];
+    newArchaeologistPrivateKeys[index] = e.target.value;
```

Comment:
It may make sense to trim the value here so that you don't need to trim it elsewhere

Reply:
Makes sense